src/ring_theory/localization/away.lean

```diff
@@ -148,4 +152,39 @@ abbreviation away_map (f : R →+* P) (r : R) :
   localization.away r →+* localization.away (f r) :=
 is_localization.away.map _ _ f r
 
+open polynomial adjoin_root
+
+/-- The `R`-`alg_hom` from the localization of `R` away from `r` to
+    `R` with an inverse of `r` adjoined. -/
+@[simps]
+noncomputable def away_to_adjoin (r : R) : away r →ₐ[R] adjoin_root (C r * X - 1) :=
+alg_hom.of_comp_eq
+  (away_lift (of (C r * X - 1)) r (is_unit_of_mul_eq_one _ (root _) (root_is_inv r)))
+  (away.away_map.lift_comp _ _)
+
+/-- The `R`-`alg_hom` from `R` with an inverse of `r` adjoined to
+    the localization of `R` away from `r`. -/
+@[simps]
+noncomputable def adjoin_to_away (r : R) : adjoin_root (C r * X - 1) →ₐ[R] away r :=
+alg_hom.of_comp_eq
+  (adjoin_root.lift (algebra_map R (away r)) (is_localization.away.inv_self r) (by simp))
+  (lift_comp_of _)
+
+@[simp] lemma away_to_adjoin_app_inv (r : R) :
+  away_to_adjoin r (is_localization.away.inv_self r) = root _ :=
+begin
+  apply inv_unique _ (root_is_inv r),
+  have : of _ r = away_to_adjoin r (algebra_map R _ r), simp,
+  rw [this, ← _root_.map_mul], simp,
+end
+
+/-- The `R`-`alg_equiv` between the localization of `R` away from `r` and
+    `R` with an inverse of `r` adjoined. -/
+noncomputable def away_equiv_adjoin (r : R) : away r ≃ₐ[R] adjoin_root (C r * X - 1) :=
```

Comment:
is it worth bundling all this together instead of having 3 different declarations?

Reply:
I think I'd like to keep `away_to_adjoin` in order to state `away_to_adjoin_app_inv`, but `adjoin_to_away` can definitely be inlined.

Reply:
can't `away_to_adjoin_app_inv` be a definition about `coe_to_fun away_equiv_adjoin`?

Reply:
Hmm let me make it a subsingleton instance and then I needn't be worried about the name...